Content.Server/Body/Metabolism/MetabolizerComponent.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Content.Shared.Chemistry.Reagent;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+
+namespace Content.Server.Body.Metabolism
+{
+    /// <summary>
+    ///     Handles metabolizing various reagents with given effects.
+    /// </summary>
+    [RegisterComponent]
+    public class MetabolizerComponent : Component
+    {
+        public override string Name => "Metabolizer";
+
+        public float AccumulatedFrametime = 0.0f;
+
+        /// <summary>
+        ///     How often to metabolize reagents, in seconds.
+        /// </summary>
+        /// <returns></returns>
+        [DataField("updateFrequency")]
+        public float UpdateFrequency = 1.0f;
+
+        /// <summary>
+        ///     Whether this metabolizer should attempt to metabolize chemicals in its parent bodies' bloodstream,
+        ///     as opposed to a solution container on the metabolizing entity itself.
+        /// </summary>
+        [DataField("takeFromBloodstream")]
+        public bool TakeFromBloodstream = true;
+
+        /// <summary>
+        ///     A dictionary mapping reagent string IDs to a list of effects & associated metabolism rate.
+        /// </summary>
+        /// <returns></returns>
+        [DataField("metabolisms", required: true)]
```

Comment:
Use `PrototypeIdDictionarySerializer` here as a custom type serializer so it validates Prototype IDs